src/engine/Service.cpp

```diff
@@ -125,9 +125,12 @@ Result Service::computeResult([[maybe_unused]] bool requestLaziness) {
       serviceQuery, "application/sparql-query",
       "application/sparql-results+json");
 
-  std::string jsonStr;
+  std::basic_string<char, std::char_traits<char>,
+                    ad_utility::AllocatorWithLimit<char>>
+      jsonStr(_executionContext->getAllocator());
   for (std::span<std::byte> bytes : jsonByteResult) {
-    absl::StrAppend(&jsonStr, reinterpret_cast<const char*>(bytes.data()));
+    jsonStr += reinterpret_cast<const char*>(bytes.data());
```

Comment:
```suggestion
    jsonStr.append(std::string_view{reinterpret_cast<const char*>(bytes.data()), bytes.size()});
```
No one guarantees you, that the `span<bytes>` is null-terminated, + this is a bit more efficient.
append is maybe necessary, because I don't know if += works with string + string_view.